eventloop/eventloop.go

```diff
@@ -24,18 +24,35 @@ func GetGlobalEventLoop() *EventLoop {
 	return GlobalEventLoop
 }
 
-func (e *EventLoop) Await(currentP *Promise) interface{} {
+func (e *EventLoop) Await(currentP *Promise) (interface{}, error) {
 	defer currentP.Done()
 	currentP.RegisterHandler()
-	val := <-currentP.rev
-	return val
+	for {
+		select {
+		case err := <-currentP.errChan:
+			return nil, err
+		case rev := <-currentP.rev:
+			return rev, nil
+		}
+	}
+
```

Comment:
The select should not be wrapped in a for-loop since we are only concerned about just one event "rev" or "errChan" once we receive data from any of the channels, data from the other channel is no longer important as data can't be received from it anymore.

Reply:
Remove the for-loop